src/components/compress/index.tsx

```diff
@@ -399,8 +419,7 @@ export default class Compress extends Component<Props, State> {
     // Either processor is good enough here.
     const processor = this.leftProcessor;
 
-    this.setState({ loadingCounter, loading: true });
-
+    this.setState({ loadingCounter, loading: true }, this.signalProcessingStart);
```

Comment:
Given that processing starts every time settings are changed, I'm not sure this is accurate. It seems like you're actually wanted an event for "file loaded".